app/src/main/java/org/wikipedia/analytics/eventplatform/ABTest.kt

```diff
@@ -0,0 +1,66 @@
+package org.wikipedia.analytics.eventplatform
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+import org.wikipedia.Constants
+import org.wikipedia.WikipediaApp
+import org.wikipedia.auth.AccountUtil
+import org.wikipedia.dataclient.ServiceFactory
+import org.wikipedia.settings.PrefsIoUtil
+import kotlin.random.Random
+
+class ABTest(private val abTestName: String, private val abTestGroupCount: Int) {
```

Comment:
I can appreciate the intention of making this a general "a/b test" class, to be reused in future tests, but most of the code here is very specific to the Machine Suggestions ABC test. Therefore, can this just be rolled into the `MachineGeneratedArticleDescriptionsAnalyticsHelper` class?

i.e. everything here can be moved to the Companion object of the `MachineGeneratedArticleDescriptionsAnalyticsHelper` class. That way, we don't even need the `machineGeneratedDescriptionsABTest` variable in the WikipediaApp class.